### PR TITLE
[SPARK-39533][ML] Deprecate scoreLabelsWeight in BinaryClassificationMetrics

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Deprecate `scoreLabelsWeight` in `BinaryClassificationMetrics`

### Why are the changes needed?

`scoreLabelsWeight` in `BinaryClassificationMetrics` is a public variable, but it should be private since it is only for internal purpose.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
existing UT

### CODE REVIEW

**Review by srowen — COMMENTED** [review]
I think this is fine. I was wondering if there is an easy way to continue to expose `scoreLabelsWeight` in case a caller is accessing it. What if it remains as a `def`? not a big deal

**Comment by zhengruifeng** [comment]
@srowen hmm, I just think it is an internal variable, should not be exposed to end users, like other priavte variables in `MulticlassMetrics` and so on. However, it had been public for a long time...

**Comment by srowen** [comment]
Right, it was still public in the binary code. If there's an easy way to retain source compatibility, that's great, otherwise OK

**Comment by zhengruifeng** [comment]
since it is only exposed in the `.mllib` side, and it is not listed in the [official documents ](https://spark.apache.org/docs/latest/mllib-evaluation-metrics.html#ranking-systems) , so I think it seems relatively safe to remove it.

may I also ask for your opinions? @huaxingao @WeichenXu123

**Comment by huaxingao** [comment]
Even though this method is not listed in the official doc, there still might be a chance a user calls this directly. I think we probably need to document this in release notes if we want to remove?

**Comment by zhengruifeng** [comment]
OK, maybe we can first deprecate it now, and remove it in the future.

File: mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala [hunk 1]
```diff
@@ -45,6 +45,8 @@ class BinaryClassificationMetrics @Since("3.0.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[_ <: Product],
     @Since("1.3.0") val numBins: Int = 1000)
   extends Logging {
+
+  @deprecated("this variable should be private and will be removed in 4.0.0.", "3.4.0")
```
Comment (huaxingao) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala:49]: Can we be more specific and say something like "The variable `scoreLabelsWeight` should be private and will be removed in 4.0.0."?

Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala:49]: sure！

**Review by huaxingao — APPROVED** [review]

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**